Python/Product/PythonTools/PythonTools/Environments/AddEnvironmentDialog.xaml.cs

```diff
@@ -143,19 +154,27 @@ enum PageKind {
             ProjectView[] projectViews;
             ProjectView selectedProjectView;
 
-            try {
-                var sln = (IVsSolution)site.GetService(typeof(SVsSolution));
-                var projects = sln?.EnumerateLoadedPythonProjects().ToArray() ?? new PythonProjectNode[0];
-
-                projectViews = projects
-                    .Select((projectNode) => new ProjectView(projectNode))
-                    .ToArray();
-
-                selectedProjectView = projectViews.SingleOrDefault(pv => pv.Node == project);
-            } catch (InvalidOperationException ex) {
-                Debug.Fail(ex.ToUnhandledExceptionMessage(typeof(AddEnvironmentDialog)));
-                projectViews = new ProjectView[0];
-                selectedProjectView = null;
+            if (workspace != null) {
+                var registryService = site.GetComponentModel().GetService<IInterpreterRegistryService>();
+                var optionsService = site.GetComponentModel().GetService<IInterpreterOptionsService>();
+                var factory = workspace.GetInterpreterFactory(registryService, optionsService);
+                selectedProjectView = new ProjectView(workspace, factory);
+                projectViews = new ProjectView[] { selectedProjectView };
+            } else {
+                try {
+                    var sln = (IVsSolution)site.GetService(typeof(SVsSolution));
+                    var projects = sln?.EnumerateLoadedPythonProjects().ToArray() ?? new PythonProjectNode[0];
```

Comment:
Array.Empty<>

Reply:
It sure what happens when null.ToArray

Reply:
if `snl?` is `null`, it should stop evaluating that expression and go to the `??` part